controllers/secret.go

```diff
@@ -80,18 +83,60 @@ func (r *OperatorPipelineReconciler) reconcileKubeConfigSecret(meta metav1.Objec
 
 // reconcileGitHubAPISecret will ensure that the GitHub API Secret is present and up to date.
 func (r *OperatorPipelineReconciler) reconcileGitHubAPISecret(meta metav1.ObjectMeta) error {
+
+	namespaceName := types.NamespacedName{
+		Namespace: meta.Namespace,
+		Name:      meta.Name,
+	}
+
+	operatorPipeline := &certv1alpha1.OperatorPipeline{}
+	err := r.Client.Get(context.Background(), namespaceName, operatorPipeline)
+	if err != nil {
+		if k8errors.IsNotFound(err) {
+			log.Info("pipeline resource not found. Ignoring since object must be deleted")
+			return nil
+		}
+
+		log.Error(err, fmt.Sprintf("unable to retrieve pipeline resource in namespace: %s ", meta.Namespace))
+		return err
+	}
+
+	var gitHubSecretName = GITHUB_API_SECRET
+	if operatorPipeline.Spec.GitHubSecretName != "" {
+		gitHubSecretName = operatorPipeline.Spec.GitHubSecretName
+	}
+
 	key := types.NamespacedName{
 		Namespace: meta.Namespace,
-		Name:      GITHUB_API_SECRET,
+		Name:      gitHubSecretName,
 	}
 
 	secret := newSecret(key)
-	if IsObjectFound(r.Client, key, secret) {
-		log.Info("existing github api secret found")
-		return nil // Existing Secret found, do nothing...
+	if !IsObjectFound(r.Client, key, secret) {
+		return errors.New(fmt.Sprintf("github api secret not found in namespace: %s", meta.Namespace))
+	}
+
+	gitHubSecret := &corev1.Secret{}
+	err = r.Get(context.Background(), key, gitHubSecret)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to retrieve github api secret in namespace: %s ", meta.Namespace))
+		return err
 	}
 
-	return errors.New("github api secret not found")
+	// checking to make sure that the key is present in the map before continue to verify the data
+	if value, ok := gitHubSecret.Data[GITHUB_TOKEN]; ok {
+		if string(value) != "" {
```

Comment:
if value is nil, I assume this might cause nil dereference. Can we check for nil first?
IMO, we better check for the erroneous condition and let the else take care of the normal condition to avoid this sort of issue.

Reply:
value will never be nil, this is checking to make sure the `GITHUB_TOKEN` is in the map before proceeding. Checking nil on a map leads to false positives since the check for the key not being found and if the key is present but the value is nil yield the same condition met.

ie 
```
if gitHubSecret.Data[GITHUB_TOKEN] == nil {
   //do stuff
}
```
Leads to false positives.

Reply:
What I meant was to check for the erroneous condition as follow:
```
if value == nil || string(value) == "" {
          // handle errors
}
else {
        // the rest of code
}
```
Checking for exceptions and errors as the first step (in the if block, rather than else block) ensures that nil dereference is handled properly.

Reply:
For this case, value would never be `nil` but I will add that check in and handle errors first.